hasktorch/src/Torch/Utils/Image.hs

```diff
@@ -31,27 +31,29 @@ readImage file = do
   dynamic_img <- I.readImage file
   case dynamic_img of
     Left err -> return $ Left err
-    Right img' -> do
-      case img' of
-        I.ImageY8 (I.Image width height vec) -> createTensor width height 1 D.UInt8 vec
-        I.ImageYA8 (I.Image width height vec) -> createTensor width height 2 D.UInt8 vec
-        I.ImageRGB8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
-        I.ImageRGBA8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
-        I.ImageYCbCr8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
-        I.ImageCMYK8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
-        _ -> return $ Left "Unsupported format"
+    Right img' -> fromDynImage img'
+
+fromDynImage :: I.DynamicImage -> IO (Either String D.Tensor)
+fromDynImage image = case image of
+  I.ImageY8 (I.Image width height vec) -> createTensor width height 1 D.UInt8 vec
+  I.ImageYA8 (I.Image width height vec) -> createTensor width height 2 D.UInt8 vec
+  I.ImageRGB8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
+  I.ImageRGBA8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
+  I.ImageYCbCr8 (I.Image width height vec) -> createTensor width height 3 D.UInt8 vec
+  I.ImageCMYK8 (I.Image width height vec) -> createTensor width height 4 D.UInt8 vec
+  _ -> return $ Left "Unsupported format, convert to 8 bit Image first"
   where
     createTensor width height channel dtype vec = do
```

Comment:
Is there a generic function for `Vec -> Tensor`? If not, should we make one?

Reply:
There is not the function. I think we should make it.